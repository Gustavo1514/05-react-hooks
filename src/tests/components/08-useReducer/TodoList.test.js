const { shallow } = require('enzyme')
const { default: TodoList } = require('../../../components/08-useReducer/TodoList')

const { demoTodos } = require('../../fixtures/demoTodos')

describe('pruebas sobre <TodoList />', () => {
  const handleDelete = jest.fn()
  const handleToogle = jest.fn()
  const wrapper = shallow(
    <TodoList
      handleDelete={handleDelete}
      handleToogle={handleToogle}
      todos={demoTodos}
    />
  )
  test('debe mostrar correctamente el componente ', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('debe listar <TodolistItem />  correctamente (el numero de elementos de demoTodos)', () => {
    //   console.log(wrapper.find('TodoListItem').length) 
      expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length)

      expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any (Function))
  });
})
