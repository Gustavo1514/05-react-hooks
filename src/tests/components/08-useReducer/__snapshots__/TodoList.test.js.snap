// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre <TodoList /> debe mostrar correctamente el componente  1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToogle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "description": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToogle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "description": "Aprender Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
